perfkitbenchmarker/centurylink/clc_virtual_machine.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ./pkb.py --cloud=Centurylink --machine_type=UBUNTU-12-64-TEMPLATE
+# --benchmarks=iperf
+
+from perfkitbenchmarker import linux_virtual_machine
+from perfkitbenchmarker import virtual_machine
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.centurylink import clc_disk
+from perfkitbenchmarker.centurylink import util
+
+
+class CenturylinkVirtualMachine(virtual_machine.BaseVirtualMachine):
+    """Object representing a Centurylink Virtual Machine."""
+
+    DEFAULT_ZONE = 'NY1'
+    DEFAULT_MACHINE_TYPE = 'UBUNTU-12-64-TEMPLATE'
+    DEFAULT_IMAGE = 'ubuntu-14-04-x64'
+
+    def __init__(self, vm_spec):
+        """Initialize a Centurylink virtual machine."""
+
+        super(CenturylinkVirtualMachine, self).__init__(vm_spec)
+        self.server_name = ''
+        self.disk_size = 1
+        self.max_local_disks = 1
+        self.local_disk_counter = 0
+
+        env = util.GetDefaultCenturylinkEnv()
+        self.api_key = env['API_KEY']
+        self.api_passwd = env['API_PASSWD']
+
+    @classmethod
+    def SetVmSpecDefaults(cls, vm_spec):
+        """Updates the VM spec with cloud specific defaults."""
+
+        if vm_spec.machine_type is None:
+            vm_spec.machine_type = cls.DEFAULT_MACHINE_TYPE
+        if vm_spec.zone is None:
+            vm_spec.zone = cls.DEFAULT_ZONE
+        if vm_spec.image is None:
+            vm_spec.image = cls.DEFAULT_IMAGE
+
+    def _Create(self):
+        """Create a Centurylink VM instance."""
+
+        env = util.GetDefaultCenturylinkEnv()
+        GROUP = env['GROUP']
+        LOCATION = env['LOCATION']
+        MACHINE_TYPE = self.machine_type
+        NAME = env['VM_NAME']
+        DESC = env['VM_DESC']
+        BACKUP_LEVEL = env['BACKUP_LEVEL']
+        NETWORK = env['NETWORK']
+        CPU = env['CPU']
+        RAM = env['RAM']
+
+        # Create a server with the specified inputs
+        create_cmd = "clc \
+            --v1-api-key %s \
+            --v1-api-passwd %s servers create \
+            --name %s \
+            --location %s \
+            --group '%s' \
+            --description '%s' \
+            --template %s \
+            --backup-level %s \
+            --network %s \
+            --cpu %d \
+            --ram %d" % (
+            self.api_key,
+            self.api_passwd,
+            NAME,
+            LOCATION,
```

Comment:
Is the "location" the same thing as a "zone"? If so, you should just use the zone passed in to the __init__ method in the vm_spec rather than using an environment variable.